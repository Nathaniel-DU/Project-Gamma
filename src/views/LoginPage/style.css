p,label {
    color: #ADEFD1FF;
    font-family: 'Sen', sans-serif;
}

.signUpbttn{
    background-color: #ADEFD1FF;
    font-family: 'Sen', sans-serif;
}
p, div {
  font-family: 'Sen', sans-serif;
  }
  
  
  .wrapper {
    height: 97vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
     background-color: #00203FFF;
  }
  
  .form-wrapper {
    display: flex;
    flex-direction: column;
    width: 280px;
    max-width: 80%;
    min-width: 100px;
    min-height: 400px;
    padding: 20px 40px;
    border-radius: 6px;
    box-shadow: 0px 8px 36px #222;
    background-color: #fefefe;
  }
  
  .form-wrapper > h2 {
    display: flex;
    justify-content: center;
    font-family: 'Sen', sans-serif;
    font-size: 2em;
    font-weight: lighter;
    margin-top: 0.25em;
    color: #222;
  }
  
  form {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
  }
  
  .info {
    padding-bottom: 1em;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }
  
  label {
    margin-bottom: 0.5em;
    color: #444;
    font-weight: lighter;
  }
  
  input {
    padding: 10px 10px;
    border-radius: 5px;
    outline: none;
    border: 1px solid #d6d1d5;
  }
  
  input::placeholder {
    font-size: 1.2em;
    font-weight: lighter;
    color: #bbb;
    font-family: 'Sen', sans-serif;
  }
  
 .login-btn {
    min-width: 90%;
    cursor: pointer;
    margin-top: 2.4em;
    padding: 	0.845em;
    border: none;
    border-radius: 10px;
    background-color: #ADEFD1FF;
    color: #000000;
    font-family: 'Sen', sans-serif;
  }
  
  button:hover {
    background-color: #4A4E69;
    color: #fefefe;
  }   
  
  .error {
    color:#db2269;
    font-size: 0.5em;
    display: relative;
  }
  
  .fullName,
  .email,
  .password {
    display: flex;
    flex-direction: column;
    margin-bottom: 15px;
  }
  
  .fullName {
    width: 100%;
  }
  
  .email,
  .password {
    width: 100%;
  }
  
  .submit {
    width: 90%;
    display: flex;
    flex-wrap: wrap;
    font-family: 'Sen', sans-serif;
  }

  #login-error {
    color: red;
    font-size: 13px;
  }