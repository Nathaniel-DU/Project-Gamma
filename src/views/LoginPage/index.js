
import React, { Component } from 'react';
import axios from 'axios';
import "./style.css"
//import {Image, Video, Transformation, CloudinaryContext} from 'cloudinary-react';

class LoginPage extends Component {
  render() {
    return (
      <Register/>
      
    );
  }
}


class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: null,
      password: null,
      loginError: ''
    };
  }
  handleChange = (event) => {
    event.preventDefault();
    const { name, value } = event.target;
    

    this.setState({[name]: value});
  }

  handleSubmit = (event) => {
    event.preventDefault();
      const user = {
        email: this.state.email,
        password: this.state.password
      }
      axios.post(`/auth/login`, user)
        .then(res => {
          if(res.data.status === 401){
            this.setState({loginError: res.data.message})
          }else{
            window.location = '/home';
          }
        })
        .catch(err => {
          console.log(err);
        })
    }



  render() {
    return (
      <div className='wrapper'>
        <div className='form-wrapper'>
          <h2>Sign In</h2>
          <form onSubmit={this.handleSubmit}>
            <div className='email'>
              <label htmlFor="email">Email</label>
              <input type='email' name='email' onChange={this.handleChange} />
            </div>
            <div className='password'>
              <label htmlFor="password">Password</label>
              <input type='password' name='password' onChange={this.handleChange} />
              <p>{this.state.loginError}</p>
              <button className="login-btn">Log In</button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
          
          
                 
    export default LoginPage;