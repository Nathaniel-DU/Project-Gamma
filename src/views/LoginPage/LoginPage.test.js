import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './index';

describe('LoginPage', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<LoginPage debug />);
        expect(component).toMatchSnapshot();
    })
})