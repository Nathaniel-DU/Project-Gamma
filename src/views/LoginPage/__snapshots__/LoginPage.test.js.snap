// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render correctly in "debug" mode 1`] = `<Register />`;
